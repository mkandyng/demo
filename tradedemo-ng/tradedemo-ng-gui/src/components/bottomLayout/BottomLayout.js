import React, {useState, useEffect} from "react";
import {connect} from "react-redux";
import { bindActionCreators } from "redux";
import {Line} from "recharts";
import {fetchIntradayTimeSeries} from "../../store/intradayTimeSeries/intradayTimeSeriesActions";
import {fetchDailyTimeSeries} from "../../store/dailyTimeSeries/dailyTimeSeriesActions";
import BottomLayoutView from "./BottomLayoutView";


/**
 * Container for the bottomLayout components
 */

function BottomLayout(props) {
    const INTRADAY_TIMESERIES_TAB_INDEX = 1;
    const DAILY_TIME_SERIES_TAB_INDEX = 2;

    const { symbol,
            orderbook,
            intradayTimeSeries,
            dailyTimeSeries,
            fetchIntradayTimeSeries,
            fetchDailyTimeSeries } = props;

    const [ selectedSymbol, updateSelectedSymbol ] = useState("");
    const [ selectedTab, updateSelectedTab ] = useState(0);

    useEffect(() => {
        updateSelectedSymbol(symbol);
        if(selectedTab === INTRADAY_TIMESERIES_TAB_INDEX) {
            fetchIntradayTimeSeries(symbol);
        } else if(selectedTab === DAILY_TIME_SERIES_TAB_INDEX) {
            fetchDailyTimeSeries(symbol);
        }
    },[symbol, selectedTab, fetchIntradayTimeSeries, fetchDailyTimeSeries])

    return (
        <BottomLayoutView symbol={selectedSymbol}
                          orderbook={orderbook}
                          intradayTimeSeries={intradayTimeSeries}
                          dailyTimeSeries={dailyTimeSeries}
                          selectedTab={selectedTab}
                          updateSelectedTab={updateSelectedTab} />
    );
}

const mapStateToProps = state => (
                        { symbol: state.marketfeed.selected.symbol,
                          orderbook: state.orderbook,
                          intradayTimeSeries: {
                              data: state.intradayTimeSeries,
                              childElements: <Line type="monotone" dataKey="price" stroke="#8884d8" activeDot={{r: 8}}/>
                          },
                          dailyTimeSeries: {
                              data: state.dailyTimeSeries,
                              childElements: [ <Line type="monotone" dataKey="open" stroke="#8884d8" activeDot={{r: 8}}/>,
                                               <Line type="monotone" dataKey="close" stroke="#82ca9d"/>]
                          }
                         });

// Map Redux actions to component props
const mapDispatchToProps = dispatch =>
    bindActionCreators({
      fetchIntradayTimeSeries,
      fetchDailyTimeSeries
    }, dispatch);

// The HOC
export default connect(mapStateToProps, mapDispatchToProps)(BottomLayout);
