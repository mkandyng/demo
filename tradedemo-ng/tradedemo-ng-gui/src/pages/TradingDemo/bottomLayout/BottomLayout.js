import React, { useState, useEffect } from "react";
import {connect} from "react-redux";
import SelectableTabs from "../../../components/SelectableTabs"
import Orderbook from "../../../modules/orderbook/Orderbook";
import TimeSeries, { TimeSeriesLines } from "../../../modules/timeSeries/TimeSeries";
import "./bottomLayout.css";

/**
 * Container for the bottomLayout components
 */
export function BottomLayout(props) {

    const { symbol,
            orders,
            intradayTimeSeries,
            dailyTimeSeries } = props

    const [ selectedSymbol, updateSelectedSymbol ] = useState("");
    const [ selectedTab, updateSelectedTab ] = useState(0);

    useEffect(() => {
        updateSelectedSymbol(symbol);
    },[symbol])

    const tabs = [
        { name: "Orderbook",
          panel: <Orderbook orders={orders} />},
        { name: "Intraday Prices " + selectedSymbol,
          panel: <TimeSeries timeSeries={intradayTimeSeries} childElements={TimeSeriesLines.INTRADAY_LINES}/>},
        { name: "Daily Prices " + selectedSymbol,
          panel: <TimeSeries timeSeries={dailyTimeSeries} childElements={TimeSeriesLines.DAILY_LINES}/>},
    ]

    return (
        <SelectableTabs containerId="bottomLayout"
                        selectedTab={selectedTab}
                        updateSelectedTab={updateSelectedTab}
                        tabs={tabs} />
    );
}

const mapStateToProps = state => (
                        { symbol: state.instruments.selected === undefined?"":state.instruments.selected.symbol,
                          orders: state.orderbook,
                          intradayTimeSeries: state.timeSeries.intradayTimeSeries,
                          dailyTimeSeries: state.timeSeries.dailyTimeSeries });

// The HOC
export default connect(mapStateToProps)(BottomLayout);
