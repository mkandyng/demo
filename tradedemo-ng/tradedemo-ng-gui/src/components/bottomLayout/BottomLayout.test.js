import React from "react";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import { UnwrapBottomLayout } from "./BottomLayout";

/**
 * This is unit test of Orderbook component
 * It leverage snapshot to test the presentation
 */

describe("UnwrapBottomLayout", () => {
    const props = { symbol: "symbol",
                    orderbook: [{symbol:"symbol"}],
                    intradayTimeSeries: {},
                    dailyTimeSeries: {},
                    fetchIntradayTimeSeries: jest.fn(),
                    fetchDailyTimeSeries: jest.fn() };


    it("should render component comparing with previous snapshot shallow markup", () => {
        // Given

        // When
        const component = shallow(<UnwrapBottomLayout {...props} />);

        // Then
        expect(toJson(component)).toMatchSnapshot();
    });

});
