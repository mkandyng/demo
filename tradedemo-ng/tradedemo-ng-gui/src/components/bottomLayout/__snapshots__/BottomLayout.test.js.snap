// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnwrapBottomLayout should render component comparing with previous snapshot shallow markup 1`] = `
<SelectableTabs
  containerId="bottomLayout"
  selectedTab={0}
  tabs={
    Array [
      Object {
        "name": "Orderbook",
        "panel": <Orderbook
          orderbook={
            Array [
              Object {
                "symbol": "symbol",
              },
            ]
          }
        />,
      },
      Object {
        "name": "Intraday Prices ",
        "panel": <TimeSeries
          timeSeries={Object {}}
        />,
      },
      Object {
        "name": "Dailyday Prices ",
        "panel": <TimeSeries
          timeSeries={Object {}}
        />,
      },
    ]
  }
  updateSelectedTab={[Function]}
/>
`;
